astropy/units/quantity_helper/function_helpers.py

```diff
@@ -182,13 +182,13 @@ def __call__(self, f=None, helps=None, module=np):
 # fmt: off
 @function_helper(
     helps={
-        np.copy, np.asfarray, np.real_if_close, np.sort_complex, np.resize,
+        np.copy, np.real_if_close, np.sort_complex, np.resize,
         np.fft.fft, np.fft.ifft, np.fft.rfft, np.fft.irfft,
         np.fft.fft2, np.fft.ifft2, np.fft.rfft2, np.fft.irfft2,
         np.fft.fftn, np.fft.ifftn, np.fft.rfftn, np.fft.irfftn,
         np.fft.hfft, np.fft.ihfft,
         np.linalg.eigvals, np.linalg.eigvalsh,
-    }
+    } | ({np.asfarray} if NUMPY_LT_2_0 else set())
```

Comment:
Okay for now but if things gets more complicated with numpy 2.1 and such, then we might need to refactor.

Reply:
I think the API is going to be a lot more stable after numpy 2.0 -- at least, that is the promise of it!